README.md

```diff
@@ -1,78 +1,405 @@
+```
+
+If the generation is taking too long, you can halt the command via <kbd>Ctrl</kbd>+<kbd>C</kbd>
+
+## Data types
+
+The generator supports four different data types:
+
+ - **integer** - any integer that can be defined by the Java [BigDecimal](https://docs.oracle.com/javase/7/docs/api/java/math/BigDecimal.html) type
+ - **decimal** - any real number (again a BigDecimal), with an optional granularity / precision that can be defined via a `granularTo` constraint.
+ - **string** - sequences of unicode characters
+ - **temporal** - specific moments in time, with values in the range 0001-01-01T00:00 to 9999-12-31T23:59
+
+<!-- TODO: rename as datetime -->
+
+We'll expand the example profile to add a new `age` field, a not-null integer in the range 1-99:
+
+~~~json
+{
+  "schemaVersion": "0.1",
+  "fields": [
+    { "name": "firstName" },
+    { "name": "age" }
+  ],
+  "rules": [
+    {
+      "constraints": [
+        { "field": "firstName", "is": "ofType", "value": "string" },
+        { "not": { "field": "firstName", "is": "null" } },
+        { "field": "firstName", "is": "matchingRegex", "value": "(Joh?n|Mar[yk])" }
+      ]
+    },
+    {
+      "constraints": [
+        { "field": "age", "is": "ofType", "value": "integer" },
+        { "field": "age", "is": "greaterThan", "value": 0 },
+        { "field": "age", "is": "lessThan", "value": 100 },
+        { "not": { "field": "age", "is": "null" } }
+      ]
+    }
+  ]
+}
+~~~
+
+With the change in the `firstname` regex, the output is now starting to look quite realistic:
+
+~~~
+firstName,age
+"John",31
+"Mary",73
+"Jon",44
+"John",24
+"Jon",43
+"Jon",18
+"Jon",5
+"John",82
+"John",19
+"Jon",43
+[...]
+~~~
+
+Finally, before exploring some more interesting features of the generator, we'll add a field for National Insurance number. In this case, the constraints applied to the field ensure that it only has a value if the age is greater than or equal to 16:
+
+~~~json
+{
+  "schemaVersion": "0.1",
+  "fields": [
+    { "name": "firstName" },
+    { "name": "age" },
+    { "name": "nationalInsurance" }
+  ],
+  "rules": [
+    {
+      "rule": "first name",
+      "constraints": [
+        { "field": "firstName", "is": "ofType", "value": "string" },
+        { "not": { "field": "firstName", "is": "null" } },
+        { "field": "firstName", "is": "matchingRegex", "value": "(Joh?n|Mar[yk])" }
+      ]
+    },
+    {
+      "rule": "age",
+      "constraints": [
+        { "field": "age", "is": "ofType", "value": "numeric" },
+        { "field": "age", "is": "greaterThan", "value": 0 },
+        { "field": "age", "is": "lessThan", "value": 100 },
+        { "not": { "field": "age", "is": "null" } }
+      ]
+    },
+    {
+      "rule": "national insurance",
+      "constraints": [
+        { "field": "nationalInsurance", "is": "ofType", "value": "string" },
+        
+        {
+          "if": {
+            "field": "age",
+            "is": "greaterThanOrEqual",
```

Comment:
Should be `"is": "greaterThanOrEqualTo"`

Reply:
well spotted!